src/main/java/reactor/netty/http/client/HttpClient.java

```diff
@@ -573,15 +575,36 @@ public final HttpClient doOnResponseError(BiConsumer<? super HttpClientResponse,
 	}
 
 	/**
-	 * Setup a callback called after {@link HttpClientResponse} has been fully received.
+	 * Setup a callback called after
+	 * {@link HttpClientState#RESPONSE_RECEIVED} has been emitted and the connection is
+	 * returned to the pool or closed.
 	 *
-	 * @param doAfterResponse a consumer observing disconnected events
+	 * @param doAfterResponse a callback called after
+	 * {@link HttpClientState#RESPONSE_RECEIVED} has been emitted and the connection is
+	 * returned to the pool or closed.
 	 *
 	 * @return a new {@link HttpClient}
+	 * @deprecated as of 0.9.5. This callback is invoked in both cases when the response is successful/there are errors.
+	 * Consider using {@link #doAfterResponseSuccess(BiConsumer)} or {@link #doOnResponseError(BiConsumer)}
```

Comment:
Maybe this can be in the main paragraph above, i.e. "Setup a callback called after... This callback is invoked in both cases when the response is successful/there..." and the then deprecation just saying in favor of...

Reply:
done